lua/autorun/server/wall_kick_parkour.lua

```diff
@@ -1,139 +1,141 @@
+                endpos = traceStartPos + ( ply:GetRight() * traceDistance * dir ),
+                filter = ply,
+            } )
+
+
+            if tr.Hit then
+                hit = true
+                accumulatedNormal = accumulatedNormal + tr.HitNormal
+                hitCount = hitCount + 1
             end
 
-            // Average all of the trace hits
-            local normal = accumulatedNormal / hitCount
-            
-            // If that trace hit something, we can start the process of kicking off
-            if hit then
-                
-                // Round the normal to coarser increments to make sure we count very similar surfaces and avoid floating point imprecision
-                local roundTo = 10
-                local x = ( math.Round( ( normal.x * 100 ) / roundTo, 0 ) * roundTo ) / 100
-                local y = ( math.Round( ( normal.y * 100 ) / roundTo, 0 ) * roundTo ) / 100
-                local z = ( math.Round( ( normal.z * 100 ) / roundTo, 0 ) * roundTo ) / 100
-                local roundedNormal = Vector( x, y, z )
-                
-                local kHeight = kickHeight or 275
-                
-                if ( not ply.LastNormal or ply.LastNormal != roundedNormal ) then
-                    local rightVel = ply:GetRight() * ply:WorldToLocal( ply:GetVelocity() + ply:GetPos()).y
-                    local upVel = Vector( 0, 0, kHeight )
-                    if ply:GetVelocity().z > kHeight then
-                        upVel = Vector( 0, 0, ( ply:GetVelocity().z - kHeight ) )
-                    end
-                    
-                    ply:ViewPunch( Angle( 0, 0, 3 * -dir ) )
-                    
-                    local totalVel = rightVel + upVel + Vector( normal.x * kickDistance, normal.y * kickDistance, 0 )
-                    
-                    ply:SetVelocity( totalVel )
-                    
-                    local footstep = footstepSounds[ math.random( 1, #footstepSounds ) ]
-                    ply:EmitSound( footstep, 100, 100, 0.25 )
-                    
-                    local snd = softSounds[ math.random( 1, #softSounds ) ]
-                    ply:EmitSound( snd )
-                    
+        end
+
+        -- Average all of the trace hits
+        local normal = accumulatedNormal / hitCount
+
+        -- If that trace hit something, we can start the process of kicking off
+        if hit then
+
+            -- Round the normal to coarser increments to make sure we count very similar surfaces and avoid floating point imprecision
+            local roundTo = 10
+            local x = ( math.Round( ( normal.x * 100 ) / roundTo, 0 ) * roundTo ) / 100
+            local y = ( math.Round( ( normal.y * 100 ) / roundTo, 0 ) * roundTo ) / 100
+            local z = ( math.Round( ( normal.z * 100 ) / roundTo, 0 ) * roundTo ) / 100
+            local roundedNormal = Vector( x, y, z )
+
+            local kHeight = kickHeight or 275
```

Comment:
This case is handled in `kickHeight`'s initialization on line 44.  If `275` is the new default value, it should be set on line 44, not here.